### PR TITLE
Add crossversion smoke tests

### PR DESCRIPTION
**Pull request by ericzolf**
DEV: add a new Vagrant configuration to do some smoke tests between the current/development version and any older one

This is something I wanted to have since a long time, but the comment from @ikus060 in PR #414 made me aware that it was time to really do something about it.

Even if you don't "speak" Ansible, the playbook-smoke-test.yml is simple to follow: `name:` describes the task and `command:` which command is being called. Can't be easier... I chose CentOS 6 as remote system because it's the basis for the oldest available manylinux, so I guess it makes interoperability as difficult as it can be.

### CODE REVIEW

**Comment by ikus060** [comment]
How to do intent to test this in our CICD pipeline ?

**Comment by ericzolf** [comment]
I don't have any plan to do so for now, and I'm not even sure how much is possible, because it would mean a VM within a VM and I'm not sure Travis supports it. It might be possible to address the requirement in a completely different way, e.g. using containers, but for now it's just for my own purpose, or any other developer's, to test locally that my changes don't break the interaction.

**Comment by ikus060** [comment]
ok, I see if it's only for development... I'm fine with what ever you do.

**Comment by ericzolf** [comment]
OK, thanks. This said, I don't deny that it would be really cool to have some cross-version checks in our pipeline.

File: tools/crossversion/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+You can use the current development version by using something like the following,
+after having built from source using `./setup.py build`:
+
+```
+PATH=../../build/scripts-3.8:$PATH PYTHONPATH=../../build/lib.linux-x86_64-3.8 \
+	ansible-playbook -v playbook-smoke-test.yml
+```
+
+> **CAUTION:** the version shown locally might be DEV or any other version
+	installed elsewhere locally, but the version used is definitely the
```
Comment (ottok) [inline tools/crossversion/README.md:31]: Codacy found an issue: [[no-blockquote-without-marker] Missing marker in block quote](https://app.codacy.com/gh/rdiff-backup/rdiff-backup/pullRequest?prid=5809598)

File: tools/crossversion/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Vagrant setup for testing older versions of rdiff-backup
+
+Just call `vagrant up` to get the version.
+
+If you want to get a different version of rdiff-backup than the default one in
+`host_vars`, use following command:
+
+```
```
Comment (ottok) [inline tools/crossversion/README.md:8]: Codacy found an issue: [[fenced-code-flag] Missing code language flag](https://app.codacy.com/gh/rdiff-backup/rdiff-backup/pullRequest?prid=5809598)

File: tools/crossversion/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+after having built from source using `./setup.py build`:
+
+```
+PATH=../../build/scripts-3.8:$PATH PYTHONPATH=../../build/lib.linux-x86_64-3.8 \
+	ansible-playbook -v playbook-smoke-test.yml
+```
+
+> **CAUTION:** the version shown locally might be DEV or any other version
+	installed elsewhere locally, but the version used is definitely the
+	built development version.
```
Comment (ottok) [inline tools/crossversion/README.md:32]: Codacy found an issue: [[no-blockquote-without-marker] Missing marker in block quote](https://app.codacy.com/gh/rdiff-backup/rdiff-backup/pullRequest?prid=5809598)

File: tools/crossversion/README.md (hunk 1 above)
Comment (ottok) [inline tools/crossversion/README.md:31]: Codacy found an issue: [[no-blockquote-without-marker] Missing marker in block quote](https://app.codacy.com/gh/rdiff-backup/rdiff-backup/pullRequest?prid=5809598)

File: tools/crossversion/README.md (hunk 2 above)
Comment (ottok) [inline tools/crossversion/README.md:8]: Codacy found an issue: [[fenced-code-flag] Missing code language flag](https://app.codacy.com/gh/rdiff-backup/rdiff-backup/pullRequest?prid=5809598)

File: tools/crossversion/README.md (hunk 3 above)
Comment (ottok) [inline tools/crossversion/README.md:32]: Codacy found an issue: [[no-blockquote-without-marker] Missing marker in block quote](https://app.codacy.com/gh/rdiff-backup/rdiff-backup/pullRequest?prid=5809598)

**Pull request merged**